bls/src/test/java/org/dashj/bls/LegacyHDKeysTest.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Copyright (c) 2022-present, Dash Core Group
+ * <p>
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+package org.dashj.bls;
+
+import org.dashj.bls.Utils.Util;
+import org.junit.Test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertThrows;
+
+public class LegacyHDKeysTest extends BaseTest {
+    @Test
+    public void shouldCreateAnExtendedPrivateKeyFromSeed() {
+        byte[] seed = new byte[]{1, 50, 6, (byte) 244, 24, (byte) 199, 1, 25};
+        ExtendedPrivateKey esk = ExtendedPrivateKey.fromSeed(seed);
+
+        ExtendedPrivateKey esk77 = esk.privateChild(77 + (1L << 31));
+        ExtendedPrivateKey esk77copy = esk.privateChild(77 + (1L << 31));
+
+        assertObjectEquals(esk77, esk77copy);
+
+        ExtendedPrivateKey esk77nh = esk.privateChild(77);
+
+        ExtendedPrivateKey eskLong = esk.privateChild((1L << 31) + 5)
+                .privateChild(0)
+                .privateChild(0)
+                .privateChild((1L << 31) + 56)
+                .privateChild(70)
+                .privateChild(4);
+        byte[] chainCode = new byte[32];
+        eskLong.getChainCode().serialize(chainCode);
+    }
+
+
+    @Test
+    public void shouldMatchDerivationThroughPrivateAndPublicKeys() {
+        byte[] seed = new byte[]{1, 50, 6, (byte) 244, 24, (byte) 199, 1, 25};
+        ExtendedPrivateKey esk = ExtendedPrivateKey.fromSeed(seed);
+        ExtendedPublicKey epk = esk.getExtendedPublicKey();
+
+        G1Element pk1 = esk.privateChild(238757).getPublicKey();
+        G1Element pk2 = epk.publicChild(238757).getPublicKey();
+
+        assertObjectEquals(pk1, pk2);
+
+        PrivateKey sk3 = esk.privateChild(0)
+                .privateChild(3)
+                .privateChild(8)
+                .privateChild(1)
+                .getPrivateKey();
+
+        G1Element pk4 = epk.publicChild(0)
+                .publicChild(3)
+                .publicChild(8)
+                .publicChild(1)
+                .getPublicKey();
+        assertObjectEquals(sk3.getG1Element(), pk4);
+
+        G2Element sig = new LegacySchemeMPL().sign(sk3, seed);
+
+        System.out.println("sig: " + Util.hexStr(sig.serialize()));
+        System.out.println("pk4: " + Util.hexStr(pk4.serialize()));
+        System.out.println("sk3: " + Util.hexStr(sk3.serialize()));
+        System.out.println("seed: " + Util.hexStr(seed));
+        //TODO: determine why this fails
+        //assertTrue(new LegacySchemeMPL().verify(sk3.getG1Element(), seed, sig));
+
+        // This part will pass, though we are using a legacy scheme above
+        // G2Element sigB = new BasicSchemeMPL().sign(sk3, seed);
+        // assertTrue(new BasicSchemeMPL().verify(sk3.getG1Element(), seed, sigB));
```

Comment:
This test fails and the reason is currently not known.  This particular feature, using `ExtendedPrivateKey` is not currently used by DashJ or Dash Wallet or DashPay, so it is not critical at this time.